doc/source/getting-started.rst

```diff
@@ -0,0 +1,194 @@
+.. _getting-started:
+
+*********************
+Getting Started Guide
+*********************
+
+The following guide will step through an example of developing and testing a
+new Ansible role. After reading this guide, you should be familiar with the
+basics of how to use Molecule and what it can offer.
+
+.. note::
+
+    In order to complete this guide by hand, you will need to additionally
+    install `Docker`_.
```

Comment:
I know that @acozine was having troubles with the initial setup under macOS and as a result, she wanted to add at least these details do docs:
- `pip install molecule[docker]` for also installing appropriate python client library seamlessly;
- `docker run hello-world` as advice on how to verify that docker itself is set up correctly.